File: llm-codegen-benchmark/baselines/springboot/src/main/java/com/benchmark/service/UserService.java
package com.benchmark.service;

import com.benchmark.model.User;
import com.benchmark.model.UserDto;
import org.springframework.stereotype.Service;

import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicLong;

@Service
public class UserService {

    private final Map<Long, User> usersDb = new ConcurrentHashMap<>();
    private final AtomicLong nextId = new AtomicLong(1);

    public User createUser(UserDto.Create request) {
        // Check for duplicate email
        boolean emailExists = usersDb.values().stream()
                .anyMatch(u -> u.getEmail().equals(request.email()));
        
        if (emailExists) {
            throw new IllegalArgumentException("Email already registered");
        }

        Long id = nextId.getAndIncrement();
        User user = new User(
                id,
                request.email(),
                request.name(),
                "hashed_" + request.password()
        );
        usersDb.put(id, user);
        return user;
    }

    public Optional<User> findById(Long id) {
        return Optional.ofNullable(usersDb.get(id));
    }
}

File: llm-codegen-benchmark/baselines/aspnetcore/src/BenchApi/Services/UserService.cs
using System.Collections.Concurrent;
using BenchApi.Models;

namespace BenchApi.Services;

public interface IUserService
{
    User CreateUser(CreateUserRequest request);
    User? FindById(long id);
}

public class UserService : IUserService
{
    private readonly ConcurrentDictionary<long, User> _usersDb = new();
    private long _nextId = 1;

    public User CreateUser(CreateUserRequest request)
    {
        // Check for duplicate email
        if (_usersDb.Values.Any(u => u.Email == request.Email))
        {
            throw new ArgumentException("Email already registered");
        }

        var id = Interlocked.Increment(ref _nextId) - 1;
        var user = new User
        {
            Id = id,
            Email = request.Email,
            Name = request.Name,
            HashedPassword = $"hashed_{request.Password}"
        };

        _usersDb[id] = user;
        return user;
    }

    public User? FindById(long id)
    {
        return _usersDb.GetValueOrDefault(id);
    }
}
